### PR TITLE
LPS-165919 portal-search-web: Prevent invoking suggestions endpoint immediately

### PR DESCRIPTION
**Pull request by oliv-yu**
https://issues.liferay.com/browse/LPS-165919 - /suggestions endpoint is invoked on each page load without actual search

Basically, this opts for a fetch function instead of relying on `useResource`, as one of its features is that it fetches upon mount. By defining a fetch function, it would only get called when the user types a new, unique value into the searchbar. Included a hook for debounce as well. Also I think `suggestionsContributorConfiguration` is usually in an array, so updated the default value for that. Let me know if there's anything else, thanks!

### CODE REVIEW

**Comment by oliv-yu** [comment]
Test failures look like they have been documented on the QA test spreadsheet, so likely unrelated... not sure what's happening with the bundle version change for kernel? Let me know if you think it's related 😅

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js [hunk 1]
```diff
@@ -35,23 +36,15 @@ export default function SearchBar({
...
 	const [active, setActive] = useState(false);
 	const [autocompleteSearchValue, setAutocompleteSearchValue] = useState('');
 	const [inputValue, setInputValue] = useState(keywords);
-	const [networkState, setNetworkState] = useState(() => ({
-		error: false,
-		loading: false,
-		networkStatus: 4,
-	}));
+	const [loading, setLoading] = useState(false);
+	const [resourceItems, setResourceItems] = useState([]);
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js:47]: Let's make `resourceItems` more specific. Perhaps renaming it to `suggestions`?

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js [hunk 2]
```diff
@@ -61,38 +54,48 @@ export default function SearchBar({
...
+			.then((data) => {
+				setLoading(false);
+
+				setResourceItems(data?.items || []);
+			})
+			.catch(() => {
+				setLoading(false);
+
+				setResourceItems([]);
 			});
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js:92]: We can move `setLoading(false);` to

```javascript
.finally(() => {
    setLoading(false);
});
```

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js [hunk 3]
```diff
@@ -61,38 +54,48 @@ export default function SearchBar({
...
-			method: 'POST',
-		},
-		fetchPolicy: 'cache-first',
-		link: fetchURL.href,
-		onNetworkStatusChange: (status) => {
-			setNetworkState({
-				error: status === 5,
-				loading: status > 1 && status < 4,
-				networkStatus: status,
+	const _handleFetchSuggestions = (value) => {
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js:57]: Let's reserve the `_handle*` keyword for functions that associate with an `on*` prop to help with identifying which functions directly associate with an event handler.

We can rename this to `_fetchSuggestions`.

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js [hunk 4]
```diff
@@ -61,38 +54,48 @@ export default function SearchBar({
...
-				: '/search',
-			groupId: Liferay.ThemeDisplay.getScopeGroupId(),
-			plid: Liferay.ThemeDisplay.getPlid(),
-			scope,
-			search: autocompleteSearchValue,
-		},
-	});
+	};
+
+	const [handleFetchSuggestionsDebounced] = useDebounceCallback(
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js:95]: Rename this as well `handleFetchSuggestionsDebounced` to `fetchSuggestionsDebounced`

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js [hunk 5]
```diff
@@ -35,23 +36,15 @@ export default function SearchBar({
 	scopeParameterStringEverything,
 	searchURL,
 	selectedEverythingSearchScope = false,
-	suggestionsContributorConfiguration = '{}',
+	suggestionsContributorConfiguration = '[]',
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js:39]: 👍

**Review by thektan — CHANGES_REQUESTED** [review]

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js (hunk 1 above)
Comment (oliv-yu) [reply modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js:47]: Oh I was wondering what to name this, but I think there is already a `suggestions` object within that data?

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js (hunk 2 above)
Comment (oliv-yu) [reply modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js:92]: Will do!

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js (hunk 1 above)
Comment (thektan) [reply modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js:47]: Maybe `suggestionsResponseData`?

Comment (thektan) [reply modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js:47]: Or `suggestionsResponseItems`

Comment (oliv-yu) [reply modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/components/SearchBar.js:47]: Thanks! I'll go with `suggestionsResponseItems` if that's all right!

**Review by thektan — APPROVED** [review]
Thanks for making those changes!

**Comment by thektan** [comment]
Failures seem unrelated to me as well.

**Comment by thektan** [comment]
ci:forward

**Comment by oliv-yu** [comment]
ci:test:relevant

**Comment by xbrianlee** [comment]
Forwarded to BChan.
https://github.com/brianchandotcom/liferay-portal/pull/126779

**Pull request closed without merging**